godel_surface_detection/src/services/surface_blending_service.cpp

```diff
@@ -247,11 +247,20 @@ void SurfaceBlendingService::save_blend_parameters(const std::string& filename)
 
 bool SurfaceBlendingService::load_path_planning_parameters(const std::string & filename)
 {
+  using godel_param_helpers::loadParam;
+
   if(godel_param_helpers::fromFile(filename, path_planning_params_))
   {
     return true;
   }
-  return false;
+  // otherwise default to the parameter server
+  ros::NodeHandle nh ("path_planning_params");
+  return  loadParam(nh, "discretization", path_planning_params_.discretization) &&
+          loadParam(nh, "margin", path_planning_params_.margin) &&
+          loadParam(nh, "overlap", path_planning_params_.overlap) &&
+          loadParam(nh, "safe_traverse_height", path_planning_params_.traverse_height) &&
+          loadParam(nh, "scan_width", path_planning_params_.scan_width) &&
+          loadParam(nh, "tool_radius", path_planning_params_.tool_radius);
```

Comment:
If any of the `loadParam` call fails the [remaining parameters will not be loaded](https://en.wikipedia.org/wiki/Short-circuit_evaluation), yet [only a warning is raised](https://github.com/tecnalia-advancedmanufacturing-robotics/godel/blob/94a63a2ad693f3d7941fe35584b481da4080ae47/godel_surface_detection/src/services/surface_blending_service.cpp#L94-L95) and initialization continues. This doesn't make a lot of sense to me.

At the very least I think we should always try to read all parameters.